analyzer/codechecker_analyzer/cmd/fixit.py

```diff
@@ -0,0 +1,299 @@
+                     characters. The full checker name must match in order to
+                     apply it.
+    file_paths -- A list of file paths to which the fixits will be applied.
+                  A file path may possibly contain * joker characters. The
+                  full path must match in order to apply it.
+    reports -- A list of CodeChecker reports. This can come from
+               "CodeChecker cmd [diff|results] ..." command in JSON format.
+    interactive -- Interactive filtering. If True then user will be asked about
+                   each fixit one by one.
+    """
+    def make_regex(parts):
+        if not parts:
+            return re.compile('.*')
+        parts = map(lambda part: re.escape(part).replace(r'\*', '.*'), parts)
+        return re.compile('|'.join(parts) + '$')
+
+    def ask_user(item):
+        print(yaml.dump(item))
+        prompt = "y/<Enter> - Yes\nOther     - No\nCtrl-C    - Cancel\nApply: "
+        return input(prompt) in "Yy"
+
+    def exists_collect(file_path):
+        if not os.path.isfile(file_path):
+            not_existing_files.add(file_path)
+            return False
+        return True
+
+    if reports:
+        def match_reports(diag_msg):
+            row, col = get_location_by_offset(diag_msg['FilePath'],
+                                              int(diag_msg['FileOffset']))
+            return any(row == report['line'] and col == report['column'] and
+                       diag_msg['FilePath'] == report['checkedFile']
+                       for report in reports)
+    else:
+        match_reports = bool
+
+    not_existing_files = set()
+    existing_files = set()
+
+    checker_names = make_regex(checker_names)
+    file_paths = make_regex(file_paths)
+
+    items = filter(
+        lambda diag: checker_names.match(diag['DiagnosticName']) and
+        len(diag['DiagnosticMessage']['Replacements']) != 0 and
+        file_paths.match(diag['DiagnosticMessage']['FilePath']) and
+        match_reports(diag['DiagnosticMessage']) and
+        exists_collect(diag['DiagnosticMessage']['FilePath']),
+        content['Diagnostics'])
+
+    if interactive:
+        items = filter(ask_user, items)
+
+    content['Diagnostics'] = list(items)
+
+    for diag in content['Diagnostics']:
+        existing_files.add(diag['DiagnosticMessage']['FilePath'])
+
+    return existing_files, not_existing_files
+
+
+def list_fixits(inputs, checker_names, file_paths, interactive, reports):
+    """
+    This function dumps the .yaml files to the standard output like a "dry run"
+    with the replacements. See clang_tidy_fixit_filter() for the documentation
+    of the filter options.
+    inputs -- A list of report directories which contains the fixit dumps in a
+              subdirectory named "fixit".
+    """
+    not_existing_files = set()
+    existing_files = set()
+
+    for i in inputs:
+        fixit_dir = os.path.join(i, 'fixit')
+
+        if not os.path.isdir(fixit_dir):
```

Comment:
Some warning message would be good to the user if this directory doesn't exist in the given input directory.